tests/http1_tests.py

```diff
@@ -367,6 +366,27 @@ def check_response(self, rsp):
         return body
 
 
+def wait_server_starting(server_port, client_port, tests, handler_cls=None):
```

Comment:
This is a factory method.  I would suggest a more "pythonic" approach would make this a class method of the TestServer class and call it everywhere we need to instantiate a TestServer.